lib/services/client_data_creator.rb

```diff
@@ -0,0 +1,41 @@
+class ClientDataCreator
+  def initialize(client_data, user, attachment_data = [])
+    @client_data = client_data
+    @user = user
+    @attachment_data = attachment_data
+  end
+
+  def run
+    client_data.build_proposal(flow: 'linear', requester: user)
```

Comment:
I don't see where in which controller "this" is done? Maybe some URLs and/or file+linenumbers would help me.